blockchain/chain/src/store/chain_store.rs

```diff
@@ -992,7 +993,9 @@ pub async fn sub_head_changes(
                 .await;
         }
     });
-    let cancel_sender = events_pubsub.write().await.subscribe();
+    let mut x = events_pubsub.write().await;
+    let cancel_sender = x.subscribe();
+    drop(x);
```

Comment:
Can't you just do this in one line? You don't need to keep the lock for anything?

Reply:
Ah yeah, i wasnt sure at one point if the lock was being held by calling subscribe, but yeah I can remove this.